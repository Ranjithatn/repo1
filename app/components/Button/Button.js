import React, { Fragment } from "react";
import Icon from "../Icon/Icon";

const Button = ({
  leftIcon,
  rightIcon,
  text,
  onClick,
  className,
  disabled = false,
  id,
  type = "button",
  style={},
}) => {
  return (
    <button
      id={id}
      type={type}
      className={"button" + (className !== undefined ? " " + className : "")}
      disabled={disabled}
      onClick={onClick}
      style={ style }
    >
      {leftIcon && <Icon icon={leftIcon} />}
      { text &&<span>{text}</span> }
      {rightIcon && <Icon icon={rightIcon} />}
    </button>
  );
};

export default Button;
