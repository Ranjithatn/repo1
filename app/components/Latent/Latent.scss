@import "../../colors.scss";
#latent {
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  .latent-top {
    display: flex;
    align-items: center;
    justify-content: enter;
    padding: 10px;
    background-color: $dark-gray;
    min-height: 45px;
    .left,
    .center,
    .right {
      flex: 1;
    }
    .center{
      // display: flex;
      justify-content: center;
    }
    .right{
      display: flex;
      justify-content: flex-end;
    }
    button {
      // position: absolute;
    }
    .title {
      color: $white;
      // margin-right: auto;
      // margin-left: auto;
    }
  }
  .latent-content {
    flex-direction: column;
    display: flex;
    flex-grow: 1;
    background-color: $dark;
    padding-top: 15%;
    overflow: auto;
    .latent-selection {
      margin-right: auto;
      margin-left: auto;
    }
    .subtitle {
      margin-top: 10px;
      margin-right: auto;
      margin-left: auto;
      color: $white;
    }
  }
}

html[dir="rtl"] {
  #latent {
    .latent-top {
      .right{
        justify-content: flex-start;
        direction: ltr;
        .button {
          .icon {
            .fa-arrow-left {
              padding-right: 20px;
            }
          }
        }
      }
    }
  }
}
