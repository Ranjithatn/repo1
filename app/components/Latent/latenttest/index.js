import React, {Component, PropTypes} from 'react';
const {dialog} = require('electron').remote

// download html2canvas and jsPDF and save the files in app/ext, or somewhere else
// the built versions are directly consumable
// import {html2canvas, jsPDF} from 'app/ext';
const html2canvas = require('html2canvas');
const jsPDF = require('jspdf')

export default class LatentTest extends Component {
  constructor(props) {
    super(props);
  }

  printDocument() {
    const input = document.getElementById('divToPrint');
    html2canvas(input)
      .then((canvas) => {
        const imgData = canvas.toDataURL('image/png');
        const pdf = new jsPDF();
        pdf.addImage(imgData, 'JPEG', 0, 0);
        // pdf.output('dataurlnewwindow');
        // var blob = pdf.output("blob");
        // window.open(URL.createObjectURL(blob),'modal');
        // window.open('', 'modal')
        // pdf.autoPrint();
        // window.open(pdf.output('dataurlnewwindow'));
        // pdf.output('datauri'); ;
        // pdf.save("download.pdf");
        dialog.showMessageBox()
//           popup.window // => Window object of new window
//           popup.browserWindow // => Remote BrowserWindow of new window
//           popup.window.document.body.innerHTML = "<h1>Hello</h1>" // You can access its DOM directly
// window.open("https://www.github.com", "github", "resizable,scrollbars,status");
      })
    ;
  }

  render() {
    return (<div>
      <div className="mb5">
        <button onClick={this.printDocument}>Print</button>
      </div>
      <div id="divToPrint" className="mt4" >
        <div>Note: Here the dimensions of div are same as A4</div> 
        <div>You Can add any component here</div>
      </div>
    </div>);
  }
}