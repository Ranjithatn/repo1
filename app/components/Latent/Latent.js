import React, { Fragment } from "react";
import { Link, withRouter } from "react-router-dom";
import { connect } from "react-redux";
// import { injectIntl } from "react-intl";

import Button from "../Button/Button";
import Title from "../Title/Title";
import Dropdown from "../Dropdown/Dropdown";
import FileBrowser from "../FileBrowser/filebrowser";
import DropdownItem from "../Dropdown/DropdownItem";
import Subtitle from "../Title/Subtitle";
import LatentImg from "../../images/latent_mock.jpeg";
import {
  requestReceivedImage,
  requestShowScannedImage,
  requestImportImage,
  requestStartLatentScan
} from "../../actions/latent";
import { requestShowNotification } from "../../actions/notifications";
import { translate } from "../../utils/intl";

import "./Latent.scss";


const findImageWidthAndHeight = (image, notification) => {
  // console.log("findImageWidthAndHeight", image);
  const img = new Image();
  img.src = image.path;
  img.onload = function() {
    // console.log("this",this);
    if ( this.width > 1600 ) {
      // console.log("display notification.");
      notification({
        message: translate("renderingImageWillTakeTime"),
        type: "is-warning"
      });
    }
  };
}


export function Latent({
  formatMessage,
  history,
  onClick,
  onChange,
  requestReceivedImage,
  requestShowScannedImage,
  requestImportImage,
  requestStartLatentScan,
  requestShowNotification,
}) {
  return (
    <Fragment>
      <div className="latent-top">
        <Button
          text={formatMessage({ id: "cancel" })}
          className="is-primary"
          id="latentclose"
          onClick={() => history.push("/authenticated/jobqueue")}
        />
        <div className="left" />
        <div className="center" style={{marginRight: "200px"}}>
          <Title is="4" text={formatMessage({ id: "title.latentInput" })} />
        </div>
      </div>
      <div className="latent-content">
        <div className="latent-selection">
          <FileBrowser
            id="importbtn"
            onChange={event => {
              findImageWidthAndHeight(event.target.files[0], requestShowNotification);
              requestReceivedImage({
                file: event.target.files[0],
                base64: ""
              });
              requestImportImage(false);
              requestStartLatentScan()
            }}
            className="is-primary is-medium"
            text={formatMessage({ id: "import" })}
            leftIcon="download"
            fileStyle={{ display: "block", height: 45 }}
          />
          &nbsp;&nbsp;&nbsp;
          <Button
            className="is-primary is-medium"
            text={formatMessage({ id: "scan" })}
            leftIcon="print"
            onClick={() => {
              requestShowScannedImage(LatentImg);
              history.push("/authenticated/latent/scanSettings");
            }}
          />
        </div>
        <Subtitle is="5" text={formatMessage({ id: "msg.noImgScanned" })} />
        <Subtitle
          is="5"
          text={formatMessage({ id: "msg.pleaseChooseImage" })}
        />
      </div>
    </Fragment>
  );
}
export default connect(null, {
  requestReceivedImage,
  requestShowScannedImage,
  requestImportImage,
  requestStartLatentScan,
  requestShowNotification,
})(withRouter(Latent));
