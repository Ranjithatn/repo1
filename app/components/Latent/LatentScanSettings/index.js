import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import CardscanSettings from "components/CardscanSettings";
import Button from "../../Button/Button";
import Title from "../../Title/Title";
import "./LatentScanSettings.scss";
import {requestStopLatentScan} from "../../../actions/latent"
export class LatentScanSettings extends Component<null, null> {
  render() {
    const {
      formatMessage,
      history,
      requestStopLatentScan
    } = this.props;
    return (
      <Fragment>
        <div className="main-top">
          <Button
            text={formatMessage({ id: "back" })}
            className="is-primary"
            onClick={() => history.push("/authenticated/latent")}
            // onClick={() =>  requestShowModal({
            //   modalType: CONFIRMATION_MODAL,
            //   modalProps: {
            //     action: () => {requestShowAuditLog("", "", "", "", 1);requestCloseLiveScanProgress()},
            //     message:"confirmationMsg"
            //   }
            // })}

          />
          <Title is="4" text={formatMessage({ id: "title.latentScan" })} />
        </div>
        <CardscanSettings
          formatMessage={formatMessage}
          type="latent"
          onCompleted={() =>{
            history.push("/authenticated/latent/latentinnov");
            requestStopLatentScan()}
          }
        />
      </Fragment>
    );
  }
}

const mapState = state => ({
});

export default connect(mapState, {
  requestStopLatentScan
})(withRouter(LatentScanSettings));
