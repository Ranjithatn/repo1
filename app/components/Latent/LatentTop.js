import React, { Fragment } from "react";
import Button from "../Button/Button";

function LatentTop({ formatMessage, history }) {
  return (
    <div className="head">
      <Button
        text={formatMessage({ id: "back" })}
        className="is-primary"
        leftIcon="arrow-left"
        onClick={() => history.goBack()}
      />&nbsp;&nbsp;&nbsp;&nbsp;
      <Button
        text={formatMessage({ id: "next" })}
        className="is-primary"
        rightIcon="arrow-right"
      />&nbsp;&nbsp;&nbsp;
    </div>
  );
}

export default LatentTop;
