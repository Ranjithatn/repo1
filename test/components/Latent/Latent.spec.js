import React from "react";
import toJson from "enzyme-to-json";
import { Latent } from "../../../app/components/Latent/Latent";
import Button from "../../../app/components/Button/Button";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";

configure({ adapter: new Adapter() });
let component, tree, instance;
let region;
const props = {
  formatMessage: data => data.id,
  // requestReceivedImage: data => {
  //   return (data.file = number), (data.base64 = "");
  // },
  requestShowScannedImage: data => data,
  // requestImportImage: boolean => boolean,
  requestImportImage:jest.fn()
};

beforeEach(() => {
  component = mount(<Latent {...props} />);
  region = component.find("FileBrowser#importbtn");

  tree = toJson(component);
  instance = component.instance();
});
describe("Latent />", () => {
  it("should render without crashing", () => {
    const component = mount(<Latent {...props} />);

    console.log("component");

    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  describe("Testing <Button /> component", () => {
    it("Check if the component Renders", () => {
      let comp = shallow(<Button />);
      expect(comp.length).toBe(1);
    });
  });
  xit("should call setAnnotationRegion when save job btn is clicked", () => {
    const evt = {
      target: { files: ["clicked"] }
    };
    region.simulate("change", false);
    expect(props.requestImportImage).toHaveBeenCalled();
  });
});
