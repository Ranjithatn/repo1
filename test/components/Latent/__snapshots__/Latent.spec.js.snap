// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Latent /> should render without crashing 1`] = `
<Latent
  formatMessage={[Function]}
  requestImportImage={[MockFunction]}
  requestShowScannedImage={[Function]}
>
  <div
    className="latent-top"
  >
    <Button
      className="is-primary"
      id="latentclose"
      onClick={[Function]}
      text="cancel"
    >
      <button
        className="button is-primary"
        disabled={false}
        id="latentclose"
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        <span>
          cancel
        </span>
      </button>
    </Button>
    <div
      className="left"
    />
    <div
      className="center"
      style={
        Object {
          "marginRight": "200px",
        }
      }
    >
      <Title
        as="h1"
        is="4"
        text="title.latentInput"
      >
        <h1
          className="title is-4"
        >
          title.latentInput
        </h1>
      </Title>
    </div>
  </div>
  <div
    className="latent-content"
  >
    <div
      className="latent-selection"
    >
      <FileBrowser
        className="is-primary is-medium"
        fileStyle={
          Object {
            "display": "block",
            "height": 45,
          }
        }
        id="importbtn"
        leftIcon="download"
        onChange={[Function]}
        text="import"
      >
        <div
          className="fileUpload button is-primary is-medium"
        >
          <Icon
            icon="download"
          >
            <span
              className="icon"
              disabled={false}
              onClick={[Function]}
              style={Object {}}
              title=""
            >
              <i
                className="fa fa-download"
              />
            </span>
          </Icon>
          <span>
            import
          </span>
          <input
            className="upload is-primary is-medium"
            id="myInput"
            onChange={[Function]}
            style={
              Object {
                "display": "block",
                "height": 45,
              }
            }
            type="file"
          />
        </div>
      </FileBrowser>
         
      <Button
        className="is-primary is-medium"
        leftIcon="print"
        onClick={[Function]}
        text="scan"
      >
        <button
          className="button is-primary is-medium"
          disabled={false}
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          <Icon
            icon="print"
          >
            <span
              className="icon"
              disabled={false}
              onClick={[Function]}
              style={Object {}}
              title=""
            >
              <i
                className="fa fa-print"
              />
            </span>
          </Icon>
          <span>
            scan
          </span>
        </button>
      </Button>
    </div>
    <Subtitle
      as="h1"
      is="5"
      text="msg.noImgScanned"
    >
      <h1
        className="subtitle is-5"
      >
        msg.noImgScanned
      </h1>
    </Subtitle>
    <Subtitle
      as="h1"
      is="5"
      text="msg.pleaseChooseImage"
    >
      <h1
        className="subtitle is-5"
      >
        msg.pleaseChooseImage
      </h1>
    </Subtitle>
  </div>
</Latent>
`;
